lib/backend/syncersv1/updateprocessors/bgpnodeprocessor.go

```diff
@@ -45,7 +45,7 @@ func (c *bgpNodeUpdateProcessor) Process(kvp *model.KVPair) ([]*model.KVPair, er
 
 	// Extract the separate bits of BGP config - these are stored as separate keys in the
 	// v1 model.  For a delete these will all be nil.
-	var asNum, ipv4, netv4, ipv6, netv6 interface{}
```

Comment:
So, theoretically we could drop the node processor code here the same way we are for the BGP peer processor and just use v3 objects in confd. As it is now, confd will be dealing with both v3 objects for BGP peers and v1 objects for bgp node data?

Don't think we need to do it now, but just making sure I understand.

Reply:
In fact what now happens is that both Node and BGPPeer resources are streamed as v3, and then:
- for the Node resources, confd calls BGPNodeUpdateProcessor.Process to convert each v3 Node to the v1 key/values that the template expects
- for the Peer resources, confd does its own thing to generate the v1 key/values that the template expects.

This was pragmatic because BGPNodeUpdateProcessor.Process was already mostly right for Nodes; it only needed addition of the rr_cluster_id field; whereas for the Peer case the handling of selectors and reverse peerings was so much new stuff that it made more sense to reimplement that from scratch.

Does that make sense?

It might make sense to move the BGPNodeUpdateProcessor code to confd, so as to make it explicit that confd is the only user of that code, and that future confd development is free to change that code if it makes sense.  WDYT?

There might be a case for moving all the update processors to the components that use them (i.e. either confd or felix), and inlining the felix syncer construction in Felix.  It feels like this would be a step along the way to eventually eliminating the v1 conversions entirely.

Reply:
sgtm!

So long as we don't end up with duplication I like the idea of slowly phasing out the v3->v1 logic by moving it to the components that use it, and then refactoring those components.